cheatsheets/gleam-for-scala-users.md

```diff
@@ -0,0 +1,1319 @@
+  // Although curly braces would work here too.
+  val y = x * (x + 10)
+  y
+}
+```
+
+or using indentation in Scala 3
+
+```scala
+def main() =
+  val x =
+    someFunction(1)
+    2
+  val y = x * (x + 10)
+  y
+```
+
+### Gleam
+
+In Gleam curly braces, `{` and `}`, are used to group expressions.
+
+```gleam
+pub fn main() {
+  let x = {
+    some_function(1)
+    2
+  }
+  // Braces are used to change precedence of arithmetic operators
+  let y = x * {x + 10}
+  y
+}
+```
+
+Like in Scala, in Gleam function blocks are always expressions, so are `case`
+blocks or arithmetic sub groups. Because they are expressions they always
+return a value.
+
+For Gleam the last value in a block's expression is always the value being
+returned from an expression.
+
+## Data types
+
+### String character encoding
+
+On the JVM, Scala uses Java's String implementation which are internally stored as ASCII
+or UTF-16 sequences. Strings can be encoded or decoded to/from a specific encoding.
+Strings are defined with double-quotes `"` or triple-double-quotes `"""`, the later
+allows for mutli-line strings.
+
+Scala has string interpolation and allows for custom string interpolators.
+
+In Gleam all strings are UTF-8 encoded binaries. Strings use double quotes and can
+span lines. Gleam strings do not allow interpolation, yet. Gleam however offers a
+`string_builder` via its standard library for performant string building.
+
+#### Scala
+
+```scala
+val what = "world"
+"""Hello
+world!"""
+s"Hellø, $what!"
+```
+
+#### Gleam
+
+```gleam
+"Hello
+world"
+
+"Hellø, world!"
+```
+
+
+### Tuples
+
+Tuples are very useful in Gleam as they're the only collection data type that
+allows mixed types in the collection. They behave very similar to Scala's tuples,
+but are 0-index instead of 1 indexed when accessing by index.
+
+#### Scala
+
+```scala
+val myTuple = ("username", "password", 10) // Type is (String, String, Int) or Tuple3[String, String, Int]
+val (_, pwd, _) = myTuple
+println(pwd) // "password"
+// Direct index access
+println(myTuple._1) // "username"
+```
+
+#### Gleam
+
+```gleam
+let my_tuple = #("username", "password", 10)
+let #(_, pwd, _) = my_tuple
+io.print(pwd) // "password"
+// Direct index access
+io.print(my_tuple.0) // "username"
+```
+
+### Lists
+
+Both Gleam and Scala use immutable linked-lists for their standard lists.
+
+#### Scala
+
+Scala's `List` is a linked-list with a cons operator (`::`).
+
+```scala
+val list = List(2, 3, 4)
+val alsoList = 2 :: 3 :: 4 :: Nil // equivalent
+val list2 = 1 :: list // list of 1,2,3,4
+val 1 :: secondElement :: tail = list2 // secondElement = 2
+```
+
+#### Gleam
+
+Gleam has a `cons` operator that works for lists destructuring and
+pattern matching. In Gleam lists are immutable so adding and removing elements
+from the start of a list is highly efficient.
+
+```gleam
+let list = [2, 3, 4]
+let list = [1, ..list]
+let [1, second_element, ..] = list
+[1.0, ..list] // compile error, type mismatch
```

Comment:
Shouldn't this example match the Scala one above?

Reply:
I've modified to Scala example to match this, as this is the example in the other cheatsheets. The last example (`[1.0, ..list] `) is a little problematic because the scala behavior is to upcast resulting list, which gets into OO type hierarchies none of which is related to the List data type